src/dymaptic.GeoBlazor.Core/Scripts/jsBuilder.ts

```diff
@@ -261,41 +261,10 @@ export function buildJsPopupTemplate(popupTemplateObject: DotNetPopupTemplate, v
 
     if (hasValue(popupTemplateObject.actions)) {
         template.actions = popupTemplateObject.actions as any;
-    }
-
-    if (viewId !== null) {
-        let view = arcGisObjectRefs[viewId] as View;
-        if (hasValue(view)) {
-            try {
-                if (hasValue(triggerActionHandler)) {
-                    triggerActionHandler.remove();
-                }
-                if (hasValue(templateTriggerActionHandler)) {
-                    templateTriggerActionHandler.remove();
-                }
-                
-                if (view.popup.on === undefined) {
-                    // we need to wait for the popup to be initialized before we can add the trigger-action handler
-                    reactiveUtils.once(() => view.popup.on !== undefined)
-                        .then(() => {
-                            templateTriggerActionHandler = view.popup.on("trigger-action", async (event: PopupTriggerActionEvent) => {
-                                await lookupDotNetRefForPopupTemplate(popupTemplateObject, viewId as string);
-                                await popupTemplateObject.dotNetPopupTemplateReference.invokeMethodAsync("OnTriggerAction", event.action.id);
-                            });
-                        })
-                } else {
-                    templateTriggerActionHandler = view.popup.on("trigger-action", async (event: PopupTriggerActionEvent) => {
-                        await lookupDotNetRefForPopupTemplate(popupTemplateObject, viewId as string);
-                        await popupTemplateObject.dotNetPopupTemplateReference.invokeMethodAsync("OnTriggerAction", event.action.id);
-                    });
-                }
-            }
-            catch (error) {
-                console.debug(error);
-            }
-        }
-    }
+    } 
 
+    popupDotNetObjects.push(popupTemplateObject);
```

Comment:
Trying to figure out the relocation of the logic.  Is it more accessible in the `arcGisJsInterop`? or was something getting lost having to move back n forth?

Reply:
This was the basis for the discussion yesterday, we are moving from a pattern of setting a `view.popup.on("trigger-action")` listener for each popup and popupTemplate, to a single listener that then can delegate in `arcGisJsInterop.setPopupHandler`